source/dofs/dof_tools.cc

```diff
@@ -1191,11 +1176,7 @@ namespace DoFTools
     std::vector<types::global_dof_index> dof_indices;
     std::vector<types::global_dof_index> dofs_on_ghosts;
 
-    typename DoFHandler<dim, spacedim>::cell_iterator cell = dof_handler.begin(
-                                                        level),
-                                                      endc =
-                                                        dof_handler.end(level);
-    for (; cell != endc; ++cell)
+    for (const auto &cell : dof_handler.active_cell_iterators())
```

Comment:
This is wrong: You're replacing a loop over one level of cells by a loop over all cells. But I think there is a function you can use to get that iterator range.